src/buffer.rs

```diff
@@ -30,6 +30,15 @@ pub enum Type {
     Private             = 0x80,
 }
 
+impl Type {
+    pub fn planar(&self) -> bool {
+        match self {
+            Type::VideoCaptureMplane | Type::VideoOutputMplane => true,
+            _ => false,
+        }
```

Comment:
Nit: there's a `matches!(self, Type::VideoCaptureMplane | Type::VideoOutputMplane)` macro for this:

```suggestion
        matches!(self, Type::VideoCaptureMplane | Type::VideoOutputMplane)
```